### PR TITLE
Turn event listeners with more than one method into to subscribers

### PR DESCRIPTION
**Pull request by leofeyer**
As briefly discussed with @Toflar in Slack.

### CODE REVIEW

**Review by bytehead — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

**Comment by aschempp** [comment]
Can you please explain the discussion?

**Comment by leofeyer** [comment]
Well, the new event listener autowiring in Symfony 4.4 assumes that event listeners have an `__invoke()` method. Of course, it is also possible to use a separate name or even multiple methods listening to different events, however, these special cases require manual container configuration and cannot be autowired – unless they are converted to event subscribers, which is what the PR does.

File: core-bundle/src/EventListener/DoctrineSchemaListener.php [hunk 1]
```diff
@@ -41,53 +37,4 @@ public function postGenerateSchema(GenerateSchemaEventArgs $event): void
     {
         $this->provider->appendToSchema($event->getSchema());
     }
-
-    /**
-     * Adds the index length on MySQL platforms for Doctrine DBAL <2.9.
-     */
-    public function onSchemaIndexDefinition(SchemaIndexDefinitionEventArgs $event): void
```
Comment (ausi) [inline core-bundle/src/EventListener/DoctrineSchemaListener.php:48]: This got removed because we now require `doctrine/dbal` in version `2.10`?

Comment (leofeyer) [reply core-bundle/src/EventListener/DoctrineSchemaListener.php:48]: Exactly.

**Comment by aschempp** [comment]
But we are not using autowiring, are we? So what's the benefit of changing *everything*?

**Comment by leofeyer** [comment]
Yes, we are:

```yaml
# BEFORE
tags:
    - { name: kernel.event_listener, method: onKernelRequest, priority: 20 }
    - { name: kernel.event_listener, method: setTranslatorLocale, priority: 100 }
```

```yaml
# AFTER
tags:
    - { name: kernel.event_subscriber }
```

**Comment by aschempp** [comment]
That's not autowiring 😂 
You're just using the "simplified" event registration, but now everything's in the class, so 👍 
Do we need the tag at all anymore, if we use autoconfiguration?

**Comment by leofeyer** [comment]
Yes, we do. At least it did not work without it. 😄

**Pull request merged**